presto-main/src/test/java/com/facebook/presto/operator/scalar/TestArrayReduceFunction.java

```diff
@@ -110,4 +110,17 @@ public void testErrorMessageOnUnknownIdentifier()
     {
         assertInvalidFunction("reduce(ARRAY [1, 2], 0, (s, x) -> s + x, s -> z)", UNKNOWN_IDENTIFIER);
     }
+
+    @Test
+    public void testErrorMessageOnWrongNumberOfArguments()
+    {
+        assertInvalidFunction("reduce(ARRAY [1, 2], 0, (s, x) -> s + x, s -> s, x -> x)", FUNCTION_NOT_FOUND,
+                "Unexpected parameters for function reduce. Expected: reduce(array(T), S, function(S,T,S), function(S,R)) for type(s): T, S, R, (wrong number of arguments: expected 4 got 5)");
+    }
+
+    @Test
+    public void testErrorMessageOnInvalidLambdaBody()
+    {
+        assertInvalidFunction("reduce(ARRAY [1, 2], 0, (s, x) -> concat(s, x), s -> s)", FUNCTION_NOT_FOUND);
```

Comment:
Added. Though, it's not the part I'm very proud of. Any suggestions for a message in this case?